json-endpoints/jetpack/class.jetpack-json-api-sync-endpoint.php

```diff
@@ -234,3 +241,144 @@ protected function result() {
 		);
 	}
 }
+
+
+class Jetpack_JSON_API_Sync_Checkout_Endpoint extends Jetpack_JSON_API_Sync_Endpoint {
+	protected function result() {
+		$args = $this->query_args();
+		$queue_name = $this->validate_queue( $args['queue'] );
+
+		if ( is_wp_error( $queue_name ) ){
+			return $queue_name;
+		}
+
+		if ( is_int( $args[ 'number_of_items' ] ) && (int) $args[ 'number_of_items' ] < 1 ) {
+			return  new WP_Error( 'invalid_number_of_items', 'Number of items needs to be in integer and should be larger that is larger then 0', 400 );
+		}
+		require_once JETPACK__PLUGIN_DIR . 'sync/class.jetpack-sync-sender.php';
+		require_once JETPACK__PLUGIN_DIR . 'sync/class.jetpack-sync-queue.php';
+		require_once JETPACK__PLUGIN_DIR . 'sync/class.jetpack-sync-json-deflate-array-codec.php';
+
+		$queue = new Jetpack_Sync_Queue( $queue_name );
+		$codec = new Jetpack_Sync_JSON_Deflate_Array_Codec();
+
+		// We need the sender to set up all the before send module actions...
+		$sender = Jetpack_Sync_Sender::get_instance();
+		
+		$skipped_items_ids = array();
+
+		// let's delete the checkin state
+		if ( isset( $args['force'] ) && $args['force'] ) {
+			$queue->force_checkin();
+		}
+
+		if ( isset( $args['show_errors'] ) && $args['show_errors'] ) {
+			error_reporting( E_ERROR | E_WARNING | E_PARSE );
+		}
+		$encode = ( $args['encode'] ? true : false );
+
+		$codec_name = $encode ? $codec->name() : null;
+
+		$buffer = $this->get_buffer( $queue, $args[ 'number_of_items' ] );
+		
+		// Check that the $buffer is not checkout out already
+		if ( is_wp_error( $buffer ) ) {
+			return new WP_Error( 'buffer-open', 'We couldn\'t get the buffer it is currently checked out', 400 );
```

Comment:
Should we use double quotes here instead of single quotes so that we don't have to escape the `'`?